halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/deploy/sizing/ComponentSizingEditCommand.java

```diff
@@ -0,0 +1,139 @@
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Parameters(separators = "=")
+public class ComponentSizingEditCommand extends AbstractConfigCommand {
+    @Getter(AccessLevel.PROTECTED)
+    private Map<String, NestableCommand> subcommands = new HashMap<>();
+
+    @Getter(AccessLevel.PUBLIC)
+    private String commandName = "edit";
+
+    @Getter(AccessLevel.PUBLIC)
+    private String serviceName;
+
+    @Parameter(
+            names = "--replicas",
+            description = "Set the number of replicas (pods) to be created for this service."
+    )
+    private Integer replicas = 1;
+
+    @Parameter(
+            names = "--pod-requests-cpu",
+            description = "Sets the cpu request for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 250m."
+    )
+    private String requestsCpu;
+
+    @Parameter(
+            names = "--pod-requests-memory",
+            description = "Sets the memory request for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 512Mi."
+    )
+    private String requestsMemory;
+
+    @Parameter(
+            names = "--pod-limits-cpu",
+            description = "Sets the cpu limit for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 1."
+    )
+    private String limitsCpu;
+
+    @Parameter(
+            names = "--pod-limits-memory",
+            description = "Sets the memory limit for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 1Gi."
+    )
+    private String limitsMemory;
+
+    public ComponentSizingEditCommand(String serviceName) {
+        this.serviceName = serviceName;
+    }
+
+    @Override
+    protected String getShortDescription() {
+        return "Edit the component sizing for service " + getServiceName();
+    }
+
+    @Override
+    protected void executeThis() {
+        String currentDeployment = getCurrentDeployment();
+        String serviceName = getServiceName();
+        DeploymentEnvironment deploymentEnvironment = new OperationHandler<DeploymentEnvironment>()
+                .setFailureMesssage("Failed to get component sizing for service " + serviceName + ".")
+                .setOperation(Daemon.getDeploymentEnvironment(currentDeployment, false))
+                .get();
+
+        CustomSizing customSizing = deploymentEnvironment.getCustomSizing();
+        int originalHash = customSizing.hashCode();
+
+        customSizing = edit(customSizing);
+
+        if (originalHash == customSizing.hashCode()) {
+            AnsiUi.failure("No changes supplied.");
+            return;
+        }
+
+        new OperationHandler<Void>()
+                .setFailureMesssage("Failed to edit component sizing for " + serviceName + ".")
```

Comment:
There are 4 lines of output messages (failure/success x get/edit) you could template with a private method.

Reply:
Not sure I'm understanding this one. Could you please help me with more details?

Reply:
Refactor out `Failed to ... ` and `Success to... ` to a common method, then call that in each 4 places. 

However I rather be it be verbose for readability sake. But I'm not strong either way.

Reply:
Agree, I also feel we'd lose on readability.

Reply:
Doesn't matter to me either way, just thought I'd call it out. 👍